src/config/mongoose_config_spec.erl

```diff
@@ -1078,6 +1094,25 @@ process_host(Host) ->
     true = Node =/= error,
     Node.
 
+process_general(General) ->
+    hosts_and_host_types_are_unique_and_non_emepty(General),
+    General.
+
+hosts_and_host_types_are_unique_and_non_emepty(General) ->
+    AllHostTypes = get_all_hosts_and_host_types(General),
+    true = lists:sort(AllHostTypes) =:= lists:usort(AllHostTypes),
+    true = [] =/= AllHostTypes.
+
+get_all_hosts_and_host_types(General) ->
+    lists:foldl(fun
+                    ({config, hosts, Hosts}, Acc) ->
+                        Acc ++ Hosts;
+                    ({config, host_types, HostTypes}, Acc) ->
+                        Acc ++ HostTypes;
```

Comment:
I think adding an include is better than this because record format is an implementation detail which can change.

Reply:
You're absolutely right, I'm just getting a bit lost in this code so didn't figure out from the first look that it's actually a record